src/rules/order.js

```diff
@@ -244,14 +244,19 @@ function getSorter(ascending) {
   const multiplier = ascending ? 1 : -1;
 
   return function importsSorter(importA, importB) {
-    let result;
-
-    if (importA < importB) {
+    let result = 0;
+
+    const A = importA.split('/');
+    const B = importB.split('/');
+    const a = A.length;
+    const b = B.length;
+    
+    if (A < B) {
```

Comment:
oh hm, i didn't think about comparing the arrays directly - i think you could certainly compare `[0]` and `[1]` this way, but i wouldn't expect this to work.

I was assuming something like:
```js
if (A.length !== B.length) {
  return B.length - A.length;
}
for (let i = 0; i < A.length) {
  if (A[i] !== B[i]) {
    return a < b ? -1 : 1;
  }
}
return 0;
```

can we add some more test cases? My hope is that the added cases will result in one of these outcomes:
1. your current approach fails them, my approach passes them
1. your current approach passes them, my approach fails them
1. both approaches pass them or fail them

Reply:
> ```js
> if (A.length !== B.length) {
>   return B.length - A.length;
> }
> ```

This will create a problem for paths like `one/two/three/four` and `one/two`



> ```js
> for (let i = 0; i < A.length) {
>   if (A[i] !== B[i]) {
>     return a < b ? -1 : 1;
>   }
> }
> ```

I believe this will work similarly to what I have added. But like you said, I'll work on a few more test cases just to be sure about both approaches.

Reply:
> i didn't think about comparing the arrays directly

Turns out it works the way we would expect. I'd still try and test both approaches just to be sure.
<img width="346" alt="image" src="https://user-images.githubusercontent.com/64613009/156896059-2ba77a17-1fcf-4b76-a5ea-81e8b2a8d4a1.png">

Reply:
`import Core from "next";
import Linka from "next/authenticate";
import Linkb from "next/authenticate/link/link2";
import Image from "next/image";
import Linkc from "next/nextauth/link";
import Auth from "next-auth";
`
I believe this is a valid test case. My code parses it correctly. 
Is there any other specific test case you have in mind?

Reply:
I suppose since the `/` is the separator, and since `/` can't appear in any of the individual parts, this works - it's basically the same as `importA < importB`, though, so can we use that to avoid the overhead of stringifying the array?

Reply:
Directly comparing strings doesn't seem to work. Splitting will be necessary I believe.

<img width="340" alt="image" src="https://user-images.githubusercontent.com/64613009/158395917-d84eabaa-5aa7-4536-9397-3000e346e2b7.png">

Reply:
hm, i suspect this is because the array joins with a comma. What about adding some test cases with commas in the parts? like `next/auth` vs `next,auth`

Reply:
> What about adding some test cases with commas in the parts?

I am confused though, commas and a few other characters are not allowed for use for filenames right?

Reply:
> i suspect this is because the array joins with a comma.

Yep definitely creates an issue. Will have go back to the previous method of iterating through the arrays instead of direct comparison.

<img width="337" alt="image" src="https://user-images.githubusercontent.com/64613009/158936859-df4f6bb3-8926-43a9-80fc-755b98d52825.png">

Reply:
Will make the changes and add related changes. Iterating through the array should work fine in my opinion.